### PR TITLE
ddl: Cancel DDL job when the key is exist

### PR DESCRIPTION
**Pull request by zimulala**
Fix issue #1829

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @shenli @qiuyesuifeng @coocood

File: ddl/ddl_db_test.go [hunk 1]
```diff
@@ -98,6 +99,79 @@ func (s *testDBSuite) testGetTable(c *C, name string) table.Table {
 	return tbl
 }
 
+func sessionExecute(s kv.Storage, sql string, done chan error) {
```
Comment (coocood) [inline ddl/ddl_db_test.go]: Rename to `backgroundExecute` is clearer.

File: model/ddl.go [hunk 2]
```diff
@@ -165,6 +165,10 @@ type JobState byte
 const (
 	JobNone JobState = iota
 	JobRunning
+	// Add unique index encounter the error of duplicated key,
+	// then convert add index to drop index to ensure to delete keys have been added.
```
Comment (coocood) [inline model/ddl.go]: When DDL encouterred an unrecoverable error at reorganization state, some keys has been added already, we need to remove them. JobRollback is the state to do rollback work.

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

File: ddl/index.go [hunk 3]
```diff
@@ -221,6 +235,23 @@ func (d *ddl) onCreateIndex(t *meta.Meta, job *model.Job) error {
 	}
 }
 
+func (d *ddl) convert2RollbackJob(t *meta.Meta, job *model.Job, tblInfo *model.TableInfo, indexInfo *model.IndexInfo) error {
+	job.State = model.JobRollback
+	job.Args = []interface{}{indexInfo.Name}
+	// If add index job rollbacks in write reorganization state, its need to delete all keys which has been added.
```
Comment (qiuyesuifeng) [inline ddl/index.go]: I have a question that, if there is a new insert sql occurs in rollback state, is there any problem?

**Comment by tiancaiamao** [comment]
LGTM

File: ddl/index.go (hunk 3 above)
Comment (zimulala) [inline ddl/index.go]: It's OK. You can treat this matter with `drop index` job. The write reorganization state in add index job that likes write only state in drop index job.  WriteOnly(WriteReorg) -> DeleteOnly -> DeleteOnly -> DeleteReorg -> Absent.

**Comment by zimulala** [comment]
PTAL @qiuyesuifeng

**Comment by zimulala** [comment]
PTAL @qiuyesuifeng

**Pull request merged**